File: OCEBook/app/src/main/java/com/devgrapher/ocebook/util/PageCounts.java
package com.devgrapher.ocebook.util;

import com.devgrapher.ocebook.model.PaginationInfo;

/**
 * Created by Brent on 2/19/17.
 */

public class PageCounts {
    private final int[] mPageCountAccumulates;
    private int mTotalCount;
    private boolean isUpdating = true;

    public PageCounts(int totalCount) {
        mPageCountAccumulates = new int[totalCount];
    }

    public void updatePage(int spineIndex, int pageCount) {
        mPageCountAccumulates[spineIndex] = mTotalCount;
        mTotalCount += pageCount;
    }

    public void updateComplete() {
        isUpdating = false;
    }

    public boolean isUpdating() {
        return isUpdating;
    }

    public int getTotalCount() {
        return mTotalCount;
    }

    public int calculateCurrentPage(int spineIndex, int spainePage) {
        if (spineIndex < mPageCountAccumulates.length) {
            return mPageCountAccumulates[spineIndex] + spainePage;
        } else {
            return 0;
        }
    }

}

File: OCEBook/app/src/main/java/com/devgrapher/ocebook/util/PageCounts.kt
package com.devgrapher.ocebook.util

/**
 * Created by Brent on 2/19/17.
 */

class PageCounts(totalCount: Int) {
    private val mPageCountAccumulates: IntArray
    var totalCount: Int = 0
        private set
    var isUpdating = true
        private set

    init {
        mPageCountAccumulates = IntArray(totalCount)
    }

    fun updatePage(spineIndex: Int, pageCount: Int) {
        mPageCountAccumulates[spineIndex] = totalCount
        totalCount += pageCount
    }

    fun updateComplete() {
        isUpdating = false
    }

    fun calculateCurrentPage(spineIndex: Int, spainePage: Int): Int {
        if (spineIndex < mPageCountAccumulates.size) {
            return mPageCountAccumulates[spineIndex] + spainePage
        } else {
            return 0
        }
    }

}
